lib/ansible/module_utils/facts/system/ssh_pub_keys.py

```diff
@@ -47,6 +47,10 @@ def collect(self, module=None, collected_facts=None):
                 key_filename = '%s/ssh_host_%s_key.pub' % (keydir, type_)
                 keydata = get_file_content(key_filename)
                 if keydata is not None:
-                    ssh_pub_key_facts[factname] = keydata.split()[1]
+                    (subtype, key) = keydata.split()
```

Comment:
This line will result in a ValueError if a key has comments, which appear after the key. The one I wrote in the other PR also has a problem if this comment is present, so we both got this wrong. We should just do
```
(subtype, key) = keydata.split()[0:2]
```
Note below how the key looks when it has a comment:
```
clint@clint-ThinkPad-X250:~$ ssh-keygen -f tmp/foo_ecdsa -t ecdsa -b 256 -C "This is my ssh key file"
Generating public/private ecdsa key pair.
Enter passphrase (empty for no passphrase): 
Enter same passphrase again: 
Your identification has been saved in tmp/foo_ecdsa.
Your public key has been saved in tmp/foo_ecdsa.pub.
The key fingerprint is:
SHA256:gM2Ebq74mo1mRpuOpC8kLfkn0LL42L6ltcLlilZhriY This is my ssh key file
The key's randomart image is:
+---[ECDSA 256]---+
|     ..          |
|    .=           |
|   .. +          |
|   oo  .         |
| +oo.   S        |
|*oooo            |
|=X=+o            |
|E^*+o.           |
|^O@*.            |
+----[SHA256]-----+
clint@clint-ThinkPad-X250:~$ cat tmp/foo_ecdsa.pub 
ecdsa-sha2-nistp256 AAAAE2VjZHNhLXNoYTItbmlzdHAyNTYAAAAIbmlzdHAyNTYAAABBBOKXJKwNtpjIfOGfgZaBX8YzV4f6QS3MOV4cb7k0hCTJckdFic5RTkz1SknN1myA5f55n/Y/WFqnvIVA3XeHs20= This is my ssh key file
```